common/src/main/java/org/opensearch/ml/common/MLModel.java

```diff
@@ -755,4 +846,191 @@ public static MLModel fromStream(StreamInput in) throws IOException {
         return new MLModel(in);
     }
 
+    public Tags getTags() {
+        return getTags(this.connector);
+    }
+
+    public Tags getTags(Connector connector) {
+        // if connector is present, model is a remote model
+        if (this.algorithm == FunctionName.REMOTE && connector != null) {
+            return getRemoteModelTags(connector);
+        }
+
+        // pre-trained models follow a specific naming convention, relying on that to identify a pre-trained model
```

Comment:
This seems can't cover the edge case that user provide model name with same pattern. For example `test/embedding/bedrock`